src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java

```diff
@@ -125,17 +132,26 @@ public Leaderboard fromString(String string) {
     };
   }
 
-  public void onLeaderboardSelected() {
-    contentPane.setVisible(false);
-    leaderboardService.getEntries(leaderboardComboBox.getValue()).thenAccept(leaderboardEntryBeans -> {
-      ratingTable.setItems(observableList(leaderboardEntryBeans));
-      contentPane.setVisible(true);
-    }).exceptionally(throwable -> {
+  public ChangeListener<Leaderboard> onLeaderboardSelected() {
+    return (obs, oldValue, newValue) -> {
       contentPane.setVisible(false);
-      log.warn("Error while loading leaderboard entries", throwable);
-      notificationService.addImmediateErrorNotification(throwable, "leaderboard.failedToLoad");
-      return null;
-    });
+      searchTextField.clear();
+      if (usernamesAutoCompletion != null) {
+        usernamesAutoCompletion.dispose();
+      }
+      leaderboardService.getEntries(newValue).thenAccept(leaderboardEntryBeans -> {
+        ratingTable.setItems(observableList(leaderboardEntryBeans));
+        usernamesAutoCompletion = TextFields.bindAutoCompletion(searchTextField,
+            leaderboardEntryBeans.stream().map(LeaderboardEntry::getUsername).collect(Collectors.toList()));
+        usernamesAutoCompletion.setDelay(0);
+        contentPane.setVisible(true);
```

Comment:
I don't see the need for the change to the listener over the onAction method

Reply:
Okay